Source/CesiumRuntime/Private/CesiumCreditSystem.cpp

```diff
@@ -163,45 +175,125 @@ void ACesiumCreditSystem::Tick(float DeltaTime) {
-    }
-    onScreenCreditString += "</body>')";
-    OnScreenCredits = UTF8_TO_TCHAR(onScreenCreditString.c_str());
-    _lastCreditsCount = creditsToShowThisFrame.size();
+    OnScreenCredits += "<credits url=\"popup\" text=\" Data attribution\"/>";
+    _creditsWidget->SetCredits(Credits, OnScreenCredits);
+  }
+  _pCreditSystem->startNextFrame();
+}
 
-    DisplayCredits = hasPopupCredits || firstScreenCredit;
+namespace {
+void convertHtmlToRtf(
+    std::string& output,
+    std::string& parentUrl,
+    TidyDoc tdoc,
+    TidyNode tnod,
+    UScreenCreditsWidget* CreditsWidget) {
+  TidyNode child;
+  for (child = tidyGetChild(tnod); child; child = tidyGetNext(child)) {
+    if (tidyNodeIsText(child)) {
+      TidyBuffer buf;
+      tidyBufInit(&buf);
+      tidyNodeGetText(tdoc, child, &buf);
+      if (buf.bp) {
+        std::string text = reinterpret_cast<const char*>(buf.bp);
+        // could not find correct option in tidy html to not add new lines
+        if (text.size() != 0 && text[text.size() - 1] == '\n') {
+          text.pop_back();
+        }
+        if (!parentUrl.empty()) {
+          output +=
+              "<credits url=\"" + parentUrl + "\"" + " text=\"" + text + "\"/>";
+        } else {
+          output += text;
+        }
+      }
+      tidyBufFree(&buf);
+    } else if (tidyNodeGetId(child) == TidyTagId::TidyTag_IMG) {
+      auto srcAttr = tidyAttrGetById(child, TidyAttrId::TidyAttr_SRC);
+      if (srcAttr) {
+        auto srcValue = tidyAttrValue(srcAttr);
+        if (srcValue) {
+          output += "<credits id=\"" +
+                    CreditsWidget->LoadImage(
+                        std::string(reinterpret_cast<const char*>(srcValue))) +
+                    "\"";
+          if (!parentUrl.empty()) {
+            output += " url=\"" + parentUrl + "\"";
+          }
+          output += "/>";
+        }
+      }
+    }
+    auto hrefAttr = tidyAttrGetById(child, TidyAttrId::TidyAttr_HREF);
+    if (hrefAttr) {
+      auto hrefValue = tidyAttrValue(hrefAttr);
+      parentUrl = std::string(reinterpret_cast<const char*>(hrefValue));
+    }
+    convertHtmlToRtf(output, parentUrl, tdoc, child, CreditsWidget);
   }
+}
+} // namespace
 
-  _pCreditSystem->startNextFrame();
+FString ACesiumCreditSystem::ConvertHtmlToRtf(std::string html) {
+  TidyDoc tdoc;
+  TidyBuffer docbuf = {0};
+  TidyBuffer tidy_errbuf = {0};
+  int err;
+
+  tdoc = tidyCreate();
+  tidyOptSetBool(tdoc, TidyForceOutput, yes);
+  tidyOptSetInt(tdoc, TidyWrapLen, 0);
+  tidyOptSetInt(tdoc, TidyNewline, TidyLF);
+
+  tidySetErrorBuffer(tdoc, &tidy_errbuf);
+  tidyBufInit(&docbuf);
+
+  // avoid buffer overflow by wrapping with html tags
+  html = "<span>" + html + "</span>";
+
+  tidyBufAppend(
+      &docbuf,
+      reinterpret_cast<void*>(const_cast<char*>(html.c_str())),
+      static_cast<uint>(html.size()));
+
+  std::string output, url;
+  err = tidyParseBuffer(tdoc, &docbuf);
+  if (err >= 0) {
```

Comment:
This doesn't look right. 0 means success, 1 means warnings, 2 means errors. I don't think it'll ever be negative.